### PR TITLE
WT-7712 commit and durable timestamps should be disallowed at stable timestamp (v5.0 backport)

### PR DESCRIPTION
**Pull request by jiechenbo**

### CODE REVIEW

File: src/docs/timestamp-txn.dox [hunk 1]
```diff
@@ -0,0 +1,155 @@
+/*! @page timestamp_txn_api Managing the transaction timestamp state
```
Comment (jiechenbo) [inline src/docs/timestamp-txn.dox:1]: @kommiharibabu , do you think we should also backport the full timestamp-txn document? The original PR only fixes up the comments in here. Let me know what you think.

File: test/suite/test_hs07.py [hunk 2]
```diff
@@ -81,14 +81,16 @@ def test_hs(self):
         ds2 = SimpleDataSet(self, uri2, 0, key_format=self.key_format, value_format="S")
         ds2.populate()
 
-        # Pin oldest and stable to timestamp 1.
-        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(1) +
-            ',stable_timestamp=' + self.timestamp_str(1))
 
         bigvalue = "aaaaa" * 100
         bigvalue2 = "ddddd" * 100
+        # Commit at timestamp 1.
```
Comment (korteland) [inline test/suite/test_hs07.py:87]: Nitpick: line break between `bigvalue2 = ...` and `# Commit at ...` would keep this closer to the original change.

File: test/suite/test_timestamp08.py [hunk 3]
```diff
@@ -105,7 +105,7 @@ def test_timestamp_api(self):
         self.session.timestamp_transaction_uint(wiredtiger.WT_TS_TXN_TYPE_COMMIT, 7)
         self.session.commit_transaction()
 
-        # Read timestamp >= oldest timestamp
+        # Read timestamp < oldest timestamp
```
Comment (korteland) [inline test/suite/test_timestamp08.py:108]: Good spot - thanks. I think this was wrong originally.

**Review by korteland — COMMENTED** [review]
A couple of comments, mostly minor stuff. Paid particular attention to `test_timestamp08` as you requested.

File: test/suite/test_timestamp08.py (hunk 3 above)
Comment (jiechenbo) [reply test/suite/test_timestamp08.py:108]: [2c84acb](https://github.com/wiredtiger/wiredtiger/pull/10088/commits/2c84acb9ab2967d5f81bc1d18107430f965ce2fb)

File: test/suite/test_hs07.py (hunk 2 above)
Comment (jiechenbo) [reply test/suite/test_hs07.py:88]: [2c84acb](https://github.com/wiredtiger/wiredtiger/pull/10088/commits/2c84acb9ab2967d5f81bc1d18107430f965ce2fb)

File: src/txn/txn_timestamp.c [hunk 4]
```diff
@@ -594,7 +599,12 @@ __wt_txn_set_commit_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t commit_ts
     } else {
         /*
          * For a prepared transaction, the commit timestamp should not be less than the prepare
-         * timestamp.
+         * timestamp. Also, the commit timestamp cannot be set before the transaction has actually
```
[the remark this answers is not in the document]
Comment (jiechenbo) [reply src/txn/txn_timestamp.c]: removed! [2c84acb](https://github.com/wiredtiger/wiredtiger/pull/10088/commits/2c84acb9ab2967d5f81bc1d18107430f965ce2fb)

File: src/txn/txn_timestamp.c [hunk 5]
```diff
@@ -575,10 +572,18 @@ __wt_txn_set_commit_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t commit_ts
               __wt_timestamp_to_string(commit_ts, ts_string[0]),
               __wt_timestamp_to_string(oldest_ts, ts_string[1]));
 
+#ifdef WT_STANDALONE_BUILD
+        if (has_stable_ts && commit_ts <= stable_ts)
+            WT_RET_MSG(session, EINVAL, "commit timestamp %s must be after the stable timestamp %s",
+              __wt_timestamp_to_string(commit_ts, ts_string[0]),
+              __wt_timestamp_to_string(stable_ts, ts_string[1]));
+#else
+        /* Don't change this error message, MongoDB servers check for it. */
```
[the remark this answers is not in the document]
Comment (jiechenbo) [reply src/txn/txn_timestamp.c]: [2c84acb](https://github.com/wiredtiger/wiredtiger/pull/10088/commits/2c84acb9ab2967d5f81bc1d18107430f965ce2fb)

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by korteland — APPROVED** [review]
LGTM!

**Pull request merged**